aws/resource_aws_pinpoint_app.go

```diff
@@ -0,0 +1,435 @@
+		Delete: resourceAwsPinpointAppDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				Computed:      true,
+				ForceNew:      true,
+				ConflictsWith: []string{"name_prefix"},
+			},
+			"name_prefix": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"application_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			//"cloudwatch_metrics_enabled": {
+			//	Type:     schema.TypeBool,
+			//	Optional: true,
+			//	Default:  false,
+			//},
+			"campaign_hook": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Set:      campaignHookHash,
+				MaxItems: 1,
+				MinItems: 0,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"lambda_function_name": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"mode": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								pinpoint.ModeDelivery,
+								pinpoint.ModeFilter,
+							}, false),
+						},
+						"web_url": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+					},
+				},
+			},
+			"limits": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Set:      campaignLimitsHash,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"daily": {
+							Type:     schema.TypeInt,
+							Optional: true,
+						},
+						"maximum_duration": {
+							Type:     schema.TypeInt,
+							Optional: true,
+						},
+						"messages_per_second": {
+							Type:     schema.TypeInt,
+							Optional: true,
+						},
+						"total": {
+							Type:     schema.TypeInt,
+							Optional: true,
+						},
+					},
+				},
+			},
+			"quiet_time": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Set:      quietTimeHash,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"end": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"start": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceAwsPinpointAppCreate(d *schema.ResourceData, meta interface{}) error {
+	pinpointconn := meta.(*AWSClient).pinpointconn
+
+	var name string
+	if v, ok := d.GetOk("name"); ok {
+		name = v.(string)
+	} else if v, ok := d.GetOk("name_prefix"); ok {
+		name = resource.PrefixedUniqueId(v.(string))
+	} else {
+		name = resource.UniqueId()
+	}
+
+	log.Printf("[DEBUG] Pinpoint create app: %s", name)
+
+	req := &pinpoint.CreateAppInput{
+		CreateApplicationRequest: &pinpoint.CreateApplicationRequest{
+			Name: aws.String(name),
+		},
+	}
+
+	output, err := pinpointconn.CreateApp(req)
+	if err != nil {
+		return fmt.Errorf("[ERROR] creating Pinpoint app: %s", err)
```

Comment:
Nitpick: Using the hclog style `[ERROR]` can be replaced with the human readable `error` as it will be placed in the middle of the error messaging from Terraform to operators

Reply:
Fixed in https://github.com/terraform-providers/terraform-provider-aws/pull/5956/commits/0b9d8ff9d5db243ccb6e43a65adcb1b36cbcc7b5